sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md

```diff
@@ -55,97 +56,46 @@ dotnet add package Azure.Storage.Blobs
-CloudBlockBlob blob = container.GetBlockBlobReference(blobName);
+In general, SAS tokens can be provided on their own to be applied as needed, or as a complete, self-authenticating URL. The legacy library allowed providing a SAS through `StorageCredentials` as well as constructing with a complete URL.
 
-if (policyName == null)
-{
-   // Create a new access policy and define its constraints.
-   // Note that the SharedAccessBlobPolicy class is used both to define the parameters of an ad hoc SAS, and
-   // to construct a shared access policy that is saved to the container's shared access policies.
-   SharedAccessBlobPolicy adHocSAS = new SharedAccessBlobPolicy()
-   {
-       // When the start time for the SAS is omitted, the start time is assumed to be the time when the storage service receives the request.
-       // Omitting the start time for a SAS that is effective immediately helps to avoid clock skew.
-       SharedAccessExpiryTime = DateTime.UtcNow.AddHours(24),
-       Permissions = SharedAccessBlobPermissions.Read | SharedAccessBlobPermissions.Write | SharedAccessBlobPermissions.Create
-   };
-
-   // Generate the shared access signature on the blob, setting the constraints directly on the signature.
-   sasBlobToken = blob.GetSharedAccessSignature(adHocSAS);
-
-   Console.WriteLine("SAS for blob (ad hoc): {0}", sasBlobToken);
-   Console.WriteLine();
-}
-else
-{
-   // Generate the shared access signature on the blob. In this case, all of the constraints for the
-   // shared access signature are specified on the container's stored access policy.
-   sasBlobToken = blob.GetSharedAccessSignature(null, policyName);
-
-   Console.WriteLine("SAS for blob (stored access policy): {0}", sasBlobToken);
-   Console.WriteLine();
-}
+```csharp
+string sasQueryString; // the provided SAS
+Uri blobLocation; // URI to a blob the SAS grants access to
+StorageCredentials credentials = new StorageCredentials(sasQueryString);
+CloudBlob blob = new CloudBlob(blobLocation, credentials);
+```
 
-// Return the URI string for the container, including the SAS token.
-return blob.Uri + sasBlobToken;
+```csharp
+Uri blobLocationWithSAS; // self-authenticating SAS URI to a blob
+CloudBlob blob = new CloudBlob(blobLocationWithSAS);
 ```
 
 v12
-```csharp
-// Create BlobSasBuilder and specify parameters
-BlobSasBuilder sasBuilder = new BlobSasBuilder()
-{
-    BlobContainerName = containerName,
-    BlobName = blobName,
-    ExpiresOn = new DateTimeOffset
-    IPRange = new SasIPRange(System.Net.IPAddress.None, System.Net.IPAddress.None)
-};
-// Set Permissions
-sasBuilder.SetPermissions(BlobSasPermissions.Read);
 
-// Create SasQueryParameters
-BlobSasQueryParameters parameters = sasBuilder.ToSasQueryParameters(sharedKeyCredential);
+The new library only supports constructing a client with a fully constructed SAS URI. Note that since client URIs are immutable once created, there is currently no way to rotate a SAS in a client.
```

Comment:
I'd rather say that "in order to rotate SAS new instance must be created" or something like that. We'd rather point user to alternative than say it's impossible.